### PR TITLE
1. 3 page types for PNI products

### LINKED ISSUE #5507: 1. Create Product Page Types
Create: 
- [ ] ProductPage (not abstract, but not creatable through the admin)
- [ ] SoftwareProductPage (inherits ProductPage)
- [ ] GeneralProductPage (inherits ProductPage)

Only create the models and maybe some of the base fields, but don't create the Orderables that are required (that's in a different ticket #5508 )

### PR DESCRIPTION
**Pull request by KalobTaulien**
Closes #5507
Related PRs/issues #5489

Criteria:
- [x] ProductPage (not abstract, but not creatable through the admin)
- [x] SoftwareProductPage (inherits ProductPage)
- [x] GeneralProductPage (inherits ProductPage)

### CODE REVIEW

**Comment by KalobTaulien** [comment]
@Pomax There are TODO's in this PR because those particular tasks will be taken care of in future tickets (tickets that already exist in this epic)

There's a ProductPage that will hold similar fields for SoftwareProductPage and GeneralProductPages. We inherit form ProductPage so we can collectively get all the products with `ProductPage.objects.all()` and to get into the specific fields we'd use `.specific` on the queryset object. This is better for grouping and filtering similar-but-kind-of-different page types.

File: network-api/networkapi/wagtailpages/migrations/0013_productpage_types.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# Generated by Django 2.2.16 on 2020-10-26 20:15
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/migrations/0013_productpage_types.py:1]: the current pni-v4-2020 only goes up to 0004, so 0013 suggests you'll want to rebase.

File: network-api/networkapi/wagtailpages/pagemodels/products.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    CloudinaryField
+)
+from networkapi.wagtailpages.pagemodels.mixin.foundation_metadata import (
+    FoundationMetadataPageMixin
+)
+
+
+class ProductPage(FoundationMetadataPageMixin, Page):
+    """
+    ProductPage is the similar page that SoftwareProductPage and
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/products.py:16]:
```suggestion
    ProductPage is the superclass that SoftwareProductPage and
```

File: network-api/networkapi/wagtailpages/translation.py [hunk 3]
```diff
@@ -249,3 +254,32 @@ class YoutubeRegretsReporterPageTR(TranslationOptions):
...
+        'worst_case',
+        'signup_requirement_explanation',
+        'how_does_it_use_data_collected',
+        'security_updates_helptext',
+        'strong_password_helptext',
+        'manage_vulnerabilities_helptext',
+        'privacy_policy_helptext',
+        'phone_number',
+        'live_chat',
+        'email',
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/translation.py:274]: let's keep this list as short as possible until the work is done, so we can add "all the fields" as part of the last "okay these are the things we'll translate" migration.

```suggestion
        'name',
        'company',
        'price',
```

**Review by Pomax — CHANGES_REQUESTED** [review]

**Review by Pomax — APPROVED** [review]

**Pull request merged**